#175636834 Set up unit testing

hervebutera opened the pull request:
#### What does this PR do?
This PR sets up packages for testing.
#### Description of Task to be completed?
-  Adds `chai` , `chai-http`, `mocha`, 
   and `nyc`  devdependencies for testing.

- Adds scripts in .travis.yml, and before-scripts for  postgresql
  and creating travis database.

- Adds function "funcToTest" and its test to ensure tests will work.

- Makes some modifications to some code formats identified as errors by running the test.
#### How should this be manually tested?
In the terminal run:
- `git clone https://git.github.com/atlp-rwanda/tech-strikers-bn-backend`
- `cd tech-strikers-bn-backend`
- `git checkout ch-set-unit-tests-175636834`
- `npm install`  to install packages used
- `npm run coverage`  to run tests and display coverage results in a table.
#### Any background context you want to provide?
- Some codes were changed as they were identified as bugs by the tests, so
attention is needed when rebasing.

- The coveralls check is 0% because not enough tests have been written for `all files` to have
a percentage.
- Other test files should be added in the `test/ ` directory.
#### What are the relevant pivotal tracker stories?
[#175636834](https://www.pivotaltracker.com/story/show/175636834)
#### Screenshots (if appropriate)
<img width="735" alt="Capture" src="https://user-images.githubusercontent.com/48241489/99942085-c1c1e500-2d77-11eb-9db5-e515b719846e.PNG">


#### Questions:

bigelio commented:
For me it's ok but try to check for those missing semicolons.

MChristian00 reviewed (approved):
Looks GREAT!

bigelio commented:
It's ok.

bigelio commented:
You have to do rebase again.

geraudirad commented on src/routes/api/users.js:
```diff
@@ -1,10 +1,11 @@
+    User.findById(req.params.id)
         .then(function(user) {
             if (!user) {
                 return res.sendStatus(401);
```
I think the status code should be 404 because we are testing for not found.
401 is unauthorized.

geraudirad commented on src/routes/api/users.js:
```diff
@@ -14,28 +15,28 @@ router.get("/user", function(req, res, next) {
+    User.findById(req.params.id)
         .then(function(user) {
             if (!user) {
                 return res.sendStatus(401);
```
Same here. It should be 404

geraudirad reviewed (changes requested):
Kindly fix hound CI errors.

MChristian00 reviewed (changes requested):
You still have many commits. Please look into that.

hervebutera commented:
> You still have many commits. Please look into that.

The commits have been merged into one as of now.

Duncanwar commented:
@hervebu Solve the conflicts and make sure Travis is passing

hervebutera committed:
chore(tests): add devdependencies for tests and check if tests work

add mocha, chai, chai-http and nyc for testing
add a test for a function "funcToTest" to know whether tests are working.

[Finishes #175636834]

hervebutera commented:
> @hervebu Solve the conflicts and make sure Travis is passing

Conflicts solved and Travis is now passing.
